pkg/controller/certificate-shim/sync.go

```diff
@@ -508,6 +508,106 @@ func certNeedsUpdate(a, b *cmapi.Certificate) bool {
 		return true
 	}
 
+	if a.Spec.RevisionHistoryLimit != b.Spec.RevisionHistoryLimit {
+		return true
+	}
+
+	aAlgorithm, bAlgorithm := cmapi.RSAKeyAlgorithm, cmapi.RSAKeyAlgorithm
+	if a.Spec.PrivateKey != nil && a.Spec.PrivateKey.Algorithm != "" {
+		aAlgorithm = a.Spec.PrivateKey.Algorithm
+	}
+
+	if b.Spec.PrivateKey != nil && b.Spec.PrivateKey.Algorithm != "" {
+		bAlgorithm = b.Spec.PrivateKey.Algorithm
+	}
+
+	if aAlgorithm != bAlgorithm {
+		return true
+	}
+
+	aEncoding, bEncoding := cmapi.PKCS1, cmapi.PKCS1
+	if a.Spec.PrivateKey != nil && a.Spec.PrivateKey.Encoding != "" {
+		aEncoding = a.Spec.PrivateKey.Encoding
+	}
+
+	if b.Spec.PrivateKey != nil && b.Spec.PrivateKey.Encoding != "" {
+		bEncoding = b.Spec.PrivateKey.Encoding
+	}
+
+	if aEncoding != bEncoding {
+		return true
+	}
+
+	aRotationPolicy, bRotationPolicy := cmapi.RotationPolicyNever, cmapi.RotationPolicyNever
+	if a.Spec.PrivateKey != nil && a.Spec.PrivateKey.RotationPolicy != "" {
+		aRotationPolicy = a.Spec.PrivateKey.RotationPolicy
+	}
+
+	if b.Spec.PrivateKey != nil && b.Spec.PrivateKey.RotationPolicy != "" {
+		bRotationPolicy = b.Spec.PrivateKey.RotationPolicy
+	}
+
+	if aRotationPolicy != bRotationPolicy {
+		return true
+	}
+
+	// for Ed25519 private key size is ignored
+	if aAlgorithm == cmapi.Ed25519KeyAlgorithm {
+		return false
+	}
+
+	aSize, bSize := 0, 0
+	if a.Spec.PrivateKey != nil && a.Spec.PrivateKey.Size != 0 {
+		aSize = a.Spec.PrivateKey.Size
+	}
+
+	if b.Spec.PrivateKey != nil && b.Spec.PrivateKey.Size != 0 {
+		bSize = b.Spec.PrivateKey.Size
+	}
+
+	switch aAlgorithm {
```

Comment:
Suggestion: at this point I think we could just do `if aSize != bSize return false`? I think at this point we should just update the cert spec to be correct- it will be up to actual certificate controllers to determine if the updated spec actually means that there is a need for re-issuance.
Let me know if you agree

Reply:
Yeah, I agree that here we just want to update the cert spec to be up to date.
I've made all the checks simpler.